core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -77,9 +76,22 @@
 public class LogRecorder extends AbstractModelObject implements Saveable {
     private volatile String name;
 
-    public final CopyOnWriteList<Target> targets = new CopyOnWriteList<>();
+    public List<Target> targets = new ArrayList<>();
```

Comment:
How breaking is this change? If it is fairly breaking should we consider making this private and adding the appropriate public api for it?

Reply:
shouldn't be breaking at all

Reply:
Formally it breaks the binary compatibility. https://wiki.eclipse.org/Evolving_Java-based_APIs_2 . I would prefer to avoid this change if possible, especially since it also creates concurrency issues (the public collection is writable from API...)

We could consider doing a breaking change, but in such case I strongly suggest creating a new field and implementing a `readResolve()` migration logic. The new field should be private also

Reply:
> I would prefer to avoid this change if possible, especially since it also creates concurrency issues (the public collection is writable from API...)

CopyOnWriteList isn't supported by JCasC, would be surprised if this was the first time that this is been an issue.

I can do that but are you sure it's needed for this case?
and any thoughts on a name for the new var? as the one that makes sense is already there

Reply:
`logTargets` 🙈 ?

Reply:
I've made the suggested changes